[SungMatt] iP

SungMatt opened the pull request:
## MattBot

🚀 Introducing **MattBot**: Your Ultimate Task Management Assistant! 📆

Are you drowning in a sea of to-dos, overwhelmed by deadlines, and struggling to keep up with your busy schedule? Say goodbye to chaos and hello to efficiency with MattBot – your new personal task management assistant!

🌟 Why Choose MattBot? 🌟

- 📌 **Seamless Task Management**: MattBot effortlessly organizes your tasks, events, and deadlines all in one place, ensuring nothing slips through the cracks.

- 🗓️ **Smart Event Scheduler**: Never miss an important meeting or appointment again! MattBot helps you schedule events.

- 🚀 **Effortless Deadlines**: Set deadlines with ease, and MattBot will keep you updated on your progress and notify you when time is running out.

- 🔍 **Advanced Search**: Quickly find the information you need with powerful search capabilities. No more digging through endless notes or calendars!

- ~~📊 Analytics & Insights: Gain valuable insights into your productivity trends with MattBot's data-driven analytics. Understand where your time goes and make improvements.~~

- ~~🔒 Security First: Your data is your business, and we take security seriously. MattBot employs state-of-the-art encryption to keep your information safe and sound.~~

- 📱 Cross-Platform Access: Access MattBot on your preferred device – whether it's your smartphone, tablet, or computer. Stay organized wherever you are!

- ~~🤖 AI-Powered Assistance: MattBot utilizes cutting-edge artificial intelligence to adapt to your unique needs, offering smart suggestions and automating repetitive tasks.~~

- ~~🎯 Goal Tracking: Set and track your goals effortlessly with MattBot's goal management feature. Achieve your dreams one step at a time!~~

🎁 Special Offer: Try MattBot today and receive a 7-day free trial of our premium features!

🚀 Don't let disorganization hold you back. Take control of your tasks, events, and deadlines with MattBot – your ultimate task management assistant!

👉 Get started now at [www.SungMatt.com/MattBot](http://www.SungMatt.com/MattBot) and experience the future of productivity! 👈

MattBot – Your Tasks, Events, and Deadlines, All in One Place! 📅✅

To do:
- [ ] Add GUI
- [ ] Conquer the world

What MattBot can't do _yet_

1. Eat Lunch
2. Eat Dinner

```java
// You can't use it elsewhere, but here's a codeblock!
while(true) {
    sleep(100);
}
```

User Testimonies
> MattBot is the best task manager made, ever!" - Abraham Lincoln

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

SungMatt committed:
Modify name of DukeBot

SungMatt committed:
Add echo functionality

SungMatt committed:
Add basic task adding functionality

SungMatt committed:
Add Task functionality

SungMatt committed:
Add marking and unmarking

SungMatt committed:
Add functionality for multiple types of events

SungMatt committed:
Fix list type behavior

SungMatt committed:
Add autotesting

SungMatt committed:
Add error finding functionality

SungMatt committed:
Add removal function

SungMatt committed:
Add enums

SungMatt committed:
Add save functionality

SungMatt committed:
Add DateTime functionality

SungMatt committed:
Make code pretty

SungMatt committed:
Restructure directories

SungMatt committed:
Add Packaging

SungMatt committed:
Merge branch 'add-gradle-support'
Add gradle support

SungMatt committed:
Add Gradle and Checkstyle

SungMatt committed:
Updated packaging structure to follow code convention

jordankanghm commented on src/main/java/mattbot/MattBot.java:
```diff
@@ -0,0 +1,166 @@
+            // Take user input in, and process user input
+            userInput = sc.nextLine();
+            String command = userInput.split(" ",2)[0];
+            switch (command) {
```
Cases should be in the same indentation as Switch!

jordankanghm commented on src/main/java/mattbot/MattBot.java:
```diff
@@ -0,0 +1,166 @@
+                            System.out.println("I've added this to your tasks: ");
+                            System.out.println(t);
+                        } catch (DateTimeParseException e) {
+                            System.out.println("Your date is invalid. It should be in the form YYYYMMDDTHHMM. An example is 20231231T2359.");
```
Is this line too long?

jordankanghm commented on src/main/java/mattbot/MattBot.java:
```diff
@@ -0,0 +1,166 @@
+                            System.out.println("I've added this to your tasks: ");
+                            System.out.println(t);
+                        } catch (DateTimeParseException e) {
+                            System.out.println("Your date is invalid. It should be in the form YYYYMMDDTHHMM. An example is 20231231T2359.");
```
Perhaps this line is too long?

chonguschonguschongus commented on src/main/java/mattbot/task/Task.java:
```diff
@@ -0,0 +1,35 @@
+package mattbot.task;
```
Good job extracting out the Task class 👍

jordankanghm commented on src/main/java/mattbot/Parser.java:
```diff
@@ -0,0 +1,71 @@
+        String[] details = saveString.split(" \\| ");
+        Task newTask;
+        try {
+            switch (details[0]) {
```
Great job implementing the coding convention for switch case!

jordankanghm commented on src/main/java/mattbot/Storage.java:
```diff
@@ -0,0 +1,84 @@
+import java.nio.file.Paths;
+
+import java.io.File;  // Import the File class
+import java.io.IOException;  // Import the IOException class to handle errors
```
Great job remembering to import specific libraries!

jordankanghm commented on src/main/java/mattbot/Storage.java:
```diff
@@ -0,0 +1,84 @@
+            Task t = aLTask.get(i);
+            toWrite.add(t.toFile());
+        }
+        try {
```
Perhaps the logic can be separated with a blank line?

chonguschonguschongus commented on src/main/java/mattbot/MattBot.java:
```diff
@@ -0,0 +1,166 @@
+package mattbot;
```
Seems like a Ui class was not extracted/ has not been done yet?

jordankanghm commented on src/main/java/mattbot/task/Deadline.java:
```diff
@@ -0,0 +1,56 @@
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.time.temporal.ChronoUnit;
+public class Deadline extends Task {
```
Perhaps there could be an empty line to separate imports from the class?

chonguschonguschongus reviewed (commented):
Ui class seems to be missing but other than that code quality and coding standard looks fantastic 😄

jordankanghm reviewed (commented):
Overall well done on adhering to coding conventions!

SungMatt committed:
Add JUnit Compatibility

SungMatt committed:
Add .jar compilation functionality

SungMatt committed:
Add Javadocs for all methods.

SungMatt committed:
Adhere to coding convention

SungMatt committed:
Add find functionality.

SungMatt committed:
Merges coding standards

SungMatt committed:
Refined Todo coding standards

SungMatt committed:
Merge branch 'Level-9'
Add find functionality.

SungMatt committed:
Adhere to style convention

SungMatt committed:
Adhere to style convention

SungMatt committed:
Add Readme

SungMatt committed:
Bug fixes

SungMatt committed:
Fix command capitalization issue

SungMatt committed:
Restructure folders, add UI scaffolding

- Move all commander level classes to command folder
- Move all UI level classes to ui folder.

Add basic UI elements

SungMatt committed:
Add avatars for UI

SungMatt committed:
Add UI elements

SungMatt committed:
Make code pretty and SLAP

SungMatt committed:
Add GUI functionality

This commit adds functionality for a GUI to exist. After many iterations,
we are proud to announce that a GUI is now functionally available!

SungMatt committed:
Fix tutorial to actual project

SungMatt committed:
Beautify GUI

SungMatt committed:
Adds Assertions

Adds key assertions to test if input state is correct during operation.
Disabled by default.

SungMatt committed:
Allow assertions to run on Gradle

SungMatt committed:
Merge pull request #2 from SungMatt/branch-A-Assertions

Add Assertions

SungMatt committed:
Correct code quality violations

Try to make methods short (under 30 LoC each) and apply SLAP where possible.

SungMatt committed:
Merge pull request #3 from SungMatt/branch-A-CodeQuality

Correct code quality violations

SungMatt committed:
Add CI test

SungMatt committed:
Add tagging functionality

This edit adds three new commands, namely "tag", "findtag" and "removetag".
"tag" allows you to add a **new** tag to the task specified.
"findtag" allows you to find tasks with this tag.
"removetag" removes a tag from a task.

Usage:
"tag <index> <tagname>"
"findtag <tagname>"
"removetag <index> <tagname>"

SungMatt committed:
Fix grammatical issue

SungMatt committed:
Add UI screenshot

SungMatt committed:
Add photo resources for the User Guide

SungMatt committed:
Add User Guide

SungMatt committed:
Update Gradle file to make only one jar

SungMatt committed:
Format User Guide

SungMatt committed:
Improve documentation readability

SungMatt committed:
Increase readability

SungMatt committed:
Merge Documentation

SungMatt committed:
Add Exceptions

SungMatt committed:
Add start message
